test/plugin/index.spec.js

```diff
@@ -24,4 +29,89 @@ describe("plugin", () => {
     // eslint-disable-next-line no-magic-numbers
     expect(pluginWithOptions.port).to.equal(3000);
   });
+
+  describe("plugin methods", () => {
+    let stats;
+    let toJson;
+    let compilation;
+    let compiler;
+    let plugin;
+
+    beforeEach(() => {
+      stats = {};
+      toJson = base.sandbox.stub().returns(stats);
+      compilation = {
+        errors: [],
+        warnings: [],
+        getStats: () => ({ toJson }),
+        tap: base.sandbox.stub(),
+        tapAsync: base.sandbox.stub() // this is us in webpack-dashboard
+      };
+      compiler = {
+        // TODO: ONLY WEBPACK4, but that's what we have in devDeps
```

Comment:
Maybe just convert this to a full explanatory comment and remove the `TODO`. I don't thing we necessarily need to run these _unit_ tests through the different versions...